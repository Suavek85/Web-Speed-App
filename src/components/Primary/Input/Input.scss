@import '../../../scss/styles';

$block: inputwrapper;
$blockMobile: modifier($block, mobile);

.inputwrapper {
  display: flex;
  align-items: center;
  margin: 30px 0;

  //change font-size of all h2 i h1 elements

  &--mobile {
    flex-direction: column;
  }
}

:export {
  block: $block;
  blockMobile: $blockMobile;
}
