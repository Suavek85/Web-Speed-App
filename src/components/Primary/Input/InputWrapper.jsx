import React from 'react'
import classNames from 'classnames/bind'
import { useMediaQuery } from 'react-responsive'
import InputUrlUI from './InputUrl/InputUrlUI'
import InputButtonWrapper from './InputButton/InputButtonWrapper'
import Error from '../Error/Error'
import styles from './Input.scss'
import { mobileWidth } from'../../../scss/mediaqueries'

export default function InputWrapper() {

  //STYLES
  const cx = classNames.bind(styles)
  const getClass = cx({
    block: true,
    blockMobile: useMediaQuery(mobileWidth),
  })

  return (
    <div className={getClass}>
      <InputUrlUI/>
      <div className={styles.btnWrapper}>
        <InputButtonWrapper />
      </div>
      <Error />
    </div>
  )
}