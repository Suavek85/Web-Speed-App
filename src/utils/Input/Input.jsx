import React from 'react'
import styles from './Input.scss'

export default function Input( { handleChange, label, ...props }) {

  return (
    <>
      { label? (<label>{ label }</label>) : null }
      <input 
        onChange={ handleChange }
        className={ styles.block }  
        {...props} />
    </>
  )
}