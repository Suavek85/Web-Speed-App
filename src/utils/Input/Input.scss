@import '../../scss/styles';

$block: signin-input;

.#{$block} {
  width: 200px;
  height: 35px;
  padding: 6px 15px;
  margin-top: 10px;
  margin-bottom: 20px;
  font-size: em(14);
  color: #748194;
  background: #f6f7f9;
  border: 0;
  border-radius: 5px;
  outline: none;
}

:export {
  block: $block;
}
