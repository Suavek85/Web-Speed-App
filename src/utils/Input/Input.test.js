import React from 'react'
import { shallow } from 'enzyme'
import Input from './Input'

describe('Input snapshot', () => {

  it('should match the snapshot', () => {
    const wrapper = shallow(<Input />)
    expect(wrapper).toMatchSnapshot()
  })

})