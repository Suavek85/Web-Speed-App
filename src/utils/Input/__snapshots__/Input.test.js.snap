// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input snapshot should match the snapshot 1`] = `
<Fragment>
  <input />
</Fragment>
`;
